File: Math/ChatRoom.cpp
#include <iostream>

using namespace std;

int main() {
    string str;
    cin >> str;
    string hel = "hello";
    int i = 0;
    for( auto s : str) {
	if(hel[i] == s ) {
	    i++;
	}
    }
    cout << (i >= 5 ? "YES" : "NO") << "\n";
	
}

File: Math/ChatRoom.go
package main

import (
	"fmt"
)

func main() {
	var str string
	fmt.Scanln(&str)
	hel := "hello"
	j := 0
	for i := 0 ; i < len(str) && j < 5 ; i++ {
		if hel[j] == str[i] {
			j++
		}
	}
	if j >= 5 {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
		
}
